zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java

```diff
@@ -347,15 +343,32 @@ void shouldStaySoftPausedAfterRestart() {
             .during(Duration.ofSeconds(5))
             .until(RecordingExporter.getRecords()::size, hasStableValue());
 
-    final Map<Integer, Long> exportedPositionsAfterSoftPause = getExportedPositions();
-
     // then
+    assertThat(
+            RecordingExporter.messageRecords()
+                .withCorrelationKey("11")
+                .withIntent(MessageIntent.PUBLISHED)
+                .limit(1)
+                .exists())
+        .isTrue();
+    assertThat(
+            RecordingExporter.messageRecords()
+                .withCorrelationKey("12")
+                .withIntent(MessageIntent.PUBLISHED)
+                .limit(1)
+                .exists())
+        .isTrue();
+
     assertThat(recordsAfterRestart).isGreaterThan(recordsBeforePause);
+
+    // all partitions should be soft paused after restart
+    Awaitility.await().untilAsserted(this::allPartitionsSoftPausedExporting);
+
     // all partitions should have -1 as exported position due to restart and the exporter positions
     // not being able to update due to soft pause.
+    final Map<Integer, Long> exportedPositionsAfterSoftPause = getExportedPositions();
```

Comment:
:x: Like we discussed before, comparing to -1 has a potential to be flaky because of snapshots. 

I would simplify the test as follows:

```
  @Test
  void shouldStaySoftPausedAfterRestart() {
    // given
    getActuator().softPause();
    client
        .newPublishMessageCommand()
        .messageName("soft-pause")
        .correlationKey("11")
        .messageId("11")
        .send()
        .join();
    final var recordsBeforePause =
        Awaitility.await()
            .atMost(Duration.ofSeconds(30))
            .during(Duration.ofSeconds(5))
            .until(RecordingExporter.getRecords()::size, hasStableValue());
    final var exportedPositionBeforeRestart = getExportedPositions();

    // when
    CLUSTER.shutdown();
    CLUSTER.start();

    client
        .newPublishMessageCommand()
        .messageName("soft-pause")
        .correlationKey("12")
        .messageId("12")
        .send()
        .join();

    final var recordsAfterRestart =
        Awaitility.await()
            .atMost(Duration.ofSeconds(30))
            .during(Duration.ofSeconds(5))
            .until(RecordingExporter.getRecords()::size, hasStableValue());

    // then
    // all partitions should be soft paused after restart
    Awaitility.await().untilAsserted(this::allPartitionsSoftPausedExporting);

    assertThat(
            RecordingExporter.messageRecords()
                .withName("soft-pause")
                .withIntent(MessageIntent.PUBLISHED)
                .limit(2)
                .count())
        .isEqualTo(2);

    assertThat(recordsAfterRestart).isGreaterThan(recordsBeforePause);

    final Map<Integer, Long> exportedPositionAfterRestart = getExportedPositions();
    exportedPositionAfterRestart.forEach(
        (partitionId, position) ->
            assertThat(position)
                .describedAs(
                    "Partition %d should not update exported position after restart", partitionId)
                .isLessThanOrEqualTo(exportedPositionBeforeRestart.get(partitionId)));
  }
```